src/Ocelot/LoadBalancer/LoadBalancers/LeastConnection.cs

```diff
@@ -2,140 +2,85 @@
 using Ocelot.Responses;
 using Ocelot.Values;
 
-namespace Ocelot.LoadBalancer.LoadBalancers
+namespace Ocelot.LoadBalancer.LoadBalancers;
+
+public class LeastConnection : ILoadBalancer
 {
-    public class LeastConnection : ILoadBalancer
+    private readonly Func<Task<List<Service>>> _services;
+    private readonly List<Lease> _leases;
+    private readonly string _serviceName;
+    private static readonly object SyncLock = new();
+
+    public LeastConnection(Func<Task<List<Service>>> services, string serviceName)
     {
-        private readonly Func<Task<List<Service>>> _services;
-        private readonly List<Lease> _leases;
-        private readonly string _serviceName;
-        private static readonly object SyncLock = new();
+        _services = services;
+        _serviceName = serviceName;
+        _leases = new List<Lease>();
+    }
 
-        public LeastConnection(Func<Task<List<Service>>> services, string serviceName)
-        {
-            _services = services;
-            _serviceName = serviceName;
-            _leases = new List<Lease>();
-        }
+    public async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
+    {
+        var services = await _services.Invoke();
 
-        public async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
+        if (services == null)
         {
-            var services = await _services.Invoke();
-
-            if (services == null)
-            {
-                return new ErrorResponse<ServiceHostAndPort>(new ServicesAreNullError($"services were null for {_serviceName}"));
-            }
-
-            if (!services.Any())
-            {
-                return new ErrorResponse<ServiceHostAndPort>(new ServicesAreEmptyError($"services were empty for {_serviceName}"));
-            }
-
-            lock (SyncLock)
-            {
-                //todo - maybe this should be moved somewhere else...? Maybe on a repeater on seperate thread? loop every second and update or something?
-                UpdateServices(services);
-
-                var leaseWithLeastConnections = GetLeaseWithLeastConnections();
-
-                _leases.Remove(leaseWithLeastConnections);
-
-                leaseWithLeastConnections = AddConnection(leaseWithLeastConnections);
-
-                _leases.Add(leaseWithLeastConnections);
-
-                return new OkResponse<ServiceHostAndPort>(new ServiceHostAndPort(leaseWithLeastConnections.HostAndPort.DownstreamHost, leaseWithLeastConnections.HostAndPort.DownstreamPort));
-            }
+            return new ErrorResponse<ServiceHostAndPort>(new ServicesAreNullError($"Services were null in {nameof(LeastConnection)} for '{_serviceName}' during {nameof(Lease)} operation!"));
         }
-
-        public void Release(ServiceHostAndPort hostAndPort)
+        else if (services.Count == 0)
```

Comment:
```else``` is not needed here

Reply:
Again, this is Tom's logic. )) But it's OK. Let's improve this.

Reply:
Fixed in favor of single if-statement. See https://github.com/ThreeMammals/Ocelot/pull/2111/commits/c4c7ae0ff2a891d64b4f81b6a6d051cc15e71211#diff-694338581b66de3c2a9c18a02e9431a9b13e1397f2532fa282cbc2751cc6fa08